mula/tests/integration/test_task_store.py

```diff
@@ -0,0 +1,196 @@
+                storage.TaskStore.name: storage.TaskStore(self.dbconn),
+                storage.PriorityQueueStore.name: storage.PriorityQueueStore(self.dbconn),
+            }
+        )
+
+        # Organisation
+        self.organisation = OrganisationFactory()
+
+    def tearDown(self):
+        models.Base.metadata.drop_all(self.dbconn.engine)
+        self.dbconn.engine.dispose()
+
+    def test_get_status_counts(self):
+        # Arrange
+        p_items = []
+        for _ in range(10):
+            p_items.append(functions.create_p_item(self.organisation.id, 1))
+
+        for p_item in p_items[:2]:
+            task = models.Task(
+                id=p_item.id,
+                hash=p_item.hash,
+                type=functions.TestModel.type,
+                scheduler_id=p_item.scheduler_id,
+                p_item=p_item,
+                status=models.TaskStatus.QUEUED,
+                modified_at=datetime.now(timezone.utc) - timedelta(hours=1),
+            )
+
+            self.mock_ctx.datastores.task_store.create_task(task)
+
+        for p_item in p_items[2:4]:
+            task = models.Task(
+                id=p_item.id,
+                hash=p_item.hash,
+                type=functions.TestModel.type,
+                scheduler_id=p_item.scheduler_id,
+                p_item=p_item,
+                status=models.TaskStatus.COMPLETED,
+                modified_at=datetime.now(timezone.utc) - timedelta(hours=4),
+            )
+
+            self.mock_ctx.datastores.task_store.create_task(task)
+
+        for p_item in p_items[4:6]:
+            task = models.Task(
+                id=p_item.id,
+                hash=p_item.hash,
+                type=functions.TestModel.type,
+                scheduler_id=p_item.scheduler_id,
+                p_item=p_item,
+                status=models.TaskStatus.FAILED,
+                modified_at=datetime.now(timezone.utc) - timedelta(hours=1),
+            )
+
+            self.mock_ctx.datastores.task_store.create_task(task)
+
+        for p_item in p_items[6:8]:
+            task = models.Task(
+                id=p_item.id,
+                hash=p_item.hash,
+                type=functions.TestModel.type,
+                scheduler_id=p_item.scheduler_id,
+                p_item=p_item,
+                status=models.TaskStatus.DISPATCHED,
+                modified_at=datetime.now(timezone.utc) - timedelta(hours=25),
+            )
+
+            self.mock_ctx.datastores.task_store.create_task(task)
+
+        for p_item in p_items[8:]:
+            task = models.Task(
+                id=p_item.id,
+                hash=p_item.hash,
+                type=functions.TestModel.type,
+                scheduler_id=p_item.scheduler_id,
+                p_item=p_item,
+                status=models.TaskStatus.DISPATCHED,
+                modified_at=datetime.now(timezone.utc) - timedelta(hours=23),
+            )
+
+            self.mock_ctx.datastores.task_store.create_task(task)
```

Comment:
I think this piece can be optimized a bit. The only difference here I see are the ranges, `status` and `modified_at` field. So you could loop over e.g.:
```python3
for r, status, modified_at in zip((range(0, 2), range(2, 4), ...), (TaskStatus.COMPLETED, TaskStatus.FAILED, ...), (some time delta, some time delta, ...)): ...
```
(this usually looks a bit better when using pytests parametrize utils)
It's also better to use constants for the `timedelta` objects, since calling `datetime.now()` on different time timestamps could produce different results (the above example already does that)